libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/binder/map/ActiveGuidanceMapBinder.kt

```diff
@@ -9,18 +10,30 @@ import com.mapbox.navigation.dropin.binder.navigationListOf
 import com.mapbox.navigation.dropin.component.camera.CameraComponent
 import com.mapbox.navigation.dropin.component.location.LocationPuck
 import com.mapbox.navigation.dropin.component.routeline.RouteLineComponent
+import com.mapbox.navigation.ui.maps.camera.NavigationCamera
+import com.mapbox.navigation.ui.maps.camera.data.MapboxNavigationViewportDataSource
 
 @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
 internal class ActiveGuidanceMapBinder(
     private val navigationViewContext: DropInNavigationViewContext,
 ) : Binder<MapView> {
 
     override fun bind(mapView: MapView): MapboxNavigationObserver {
+        val viewPortDataSource = MapboxNavigationViewportDataSource(
+            mapboxMap = mapView.getMapboxMap()
+        )
+        val navigationCamera = NavigationCamera(
+            cameraPlugin = mapView.camera,
+            mapboxMap = mapView.getMapboxMap(),
+            viewportDataSource = viewPortDataSource
+        )
```

Comment:
Since both ...DataSource and NavigationCamera requires mapView instance, why not move this inside CameraComponent?

Reply:
I had it that way earlier, but chose to inject them and make it easy to mock for testing

Reply:
Considering the goal with the `Component` implementations is to make them easy to "drop in". Another approach to allow us to remove these classes from the constructor is to have static function providers. That is what I did for `AudioGuidance`

``` kotlin
val audioGuidanceApi = AudioGuidanceApi.create(mapboxNavigation, AudioGuidanceServices())
```

We can iterate on the implementation details below without requiring new component interfaces.